front/pages/api/w/[wId]/assistant/agent_configurations/index.ts

```diff
@@ -445,3 +445,38 @@ export async function createOrUpgradeAgentConfiguration({
 
   return new Ok(agentConfiguration);
 }
+
+function getDsvIdsFromActions(
+  actions: PostOrPatchAgentConfigurationRequestBody["assistant"]["actions"]
+): string[] {
+  const relevantActions = actions.filter(
+    (action) =>
+      action.type === "retrieval_configuration" ||
+      action.type === "process_configuration" ||
+      action.type === "tables_query_configuration"
+  );
+
+  return removeNulls(
+    relevantActions.flatMap((action) => {
+      if (action.type === "retrieval_configuration") {
+        return action.dataSources.map(
+          (dataSource) => dataSource.dataSourceViewId
+        );
+      } else if (action.type === "process_configuration") {
+        return action.dataSources.map(
+          (dataSource) => dataSource.dataSourceViewId
+        );
+      } else if (action.type === "tables_query_configuration") {
+        return action.tables.map((table) => {
+          if (!("dataSourceViewId" in table)) {
```

Comment:
`dataSourceView` should always be defined since I shipped https://github.com/dust-tt/dust/pull/7076.

Reply:
:pray:

Reply:
actually, IMO still need it until you remove the todo here:
```
export type TableDataSourceConfiguration = {
  dataSourceId: string;
  // TODO(GROUPS_INFRA) Make `dataSourceViewId` required.
  dataSourceViewId?: string | null;
  tableId: string;
  workspaceId: string;
};
```